sympy/core/tests/test_arit.py

```diff
@@ -1898,38 +1898,38 @@ def test_mul_coeff():
 
 
 def test_mul_zero_detection():
-    nz = Dummy(real=True, zero=False, finite=True)
-    r = Dummy(real=True)
-    c = Dummy(real=False, complex=True, finite=True)
-    c2 = Dummy(real=False, complex=True, finite=True)
-    i = Dummy(imaginary=True, finite=True)
+    nz = Dummy(real=True, zero=False)
+    r = Dummy(extended_real=True)
+    c = Dummy(extended_real=False, complex=True)
+    c2 = Dummy(extended_real=False, complex=True)
```

Comment:
Why have these become extended_real rather than real. If complex implies finite then e.g. `extended_real=False,complex=True` should be the same as `real=False,complex=True` since the difference between real and extended_real is the infinities that are excluded by complex=True under this PR.